src/Account/Session/ShopSession.php

```diff
@@ -169,23 +184,31 @@ protected function getAccessToken($shopUid)
     }
 
     /**
-     * @return string
-     */
-    private function getShopUuid()
-    {
-        return $this->linkShop->getShopUuid();
-    }
-
-    /**
-     * @throws InconsistentAssociationStateException
+     * @param int $waitForOAuth2ClientSeconds
      *
      * @return void
+     *
+     * @throws InconsistentAssociationStateException
      */
-    public function assertAssociationState()
+    protected function assertAssociationState($waitForOAuth2ClientSeconds = 60)
```

Comment:
60 seconds is a lifetime for an Apache server with default PHP settings (https://www.php.net/manual/en/function.set-time-limit.php).

I would suggest working on a synchronous verification workflow instead of trying to hack things this way.

Reply:
Maybe the term "wait" is not clear enough. Here we are obviously not hanging for 60 secs. We are just checking how much time elapsed since association occured.

Reply:
"timeout" would more appropriate then